From 9d466bda1837e0bfe1479df77418fc5d5e5ee92f Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:02:44 -0400
Subject: [fs] btrfs: kmalloc() doesn't return an ERR_PTR

Message-id: <1407179109-13714-278-git-send-email-sandeen@redhat.com>
Patchwork-id: 87312
O-Subject: [RHEL7.1 PATCH 277/422] Btrfs: kmalloc() doesn't return an ERR_PTR
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Dan Carpenter <dan.carpenter@oracle.com>

commit 84dbeb87d1439d3a6614c95e82429542434bafb9
Author: Dan Carpenter <dan.carpenter@oracle.com>
Date:   Fri Mar 28 11:06:00 2014 +0300

    Btrfs: kmalloc() doesn't return an ERR_PTR

    The error handling was copy and pasted from memdup_user().  It should be
    checking for NULL obviously.

    Fixes: abccd00f8af2 ('btrfs: Fix 32/64-bit problem with BTRFS_SET_RECEIVED_SUBVOL ioctl')
    Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/ioctl.c b/fs/btrfs/ioctl.c
index 0401397..27d551c 100644
--- a/fs/btrfs/ioctl.c
+++ b/fs/btrfs/ioctl.c
@@ -4531,9 +4531,8 @@ static long btrfs_ioctl_set_received_subvol_32(struct file *file,
  }
 
  args64 = kmalloc(sizeof(*args64), GFP_NOFS);
- if (IS_ERR(args64)) {
-  ret = PTR_ERR(args64);
-  args64 = NULL;
+ if (!args64) {
+  ret = -ENOMEM;
   goto out;
  }
 
-- 
1.7.1